### PR TITLE
Fixing issue 742

### PR DESCRIPTION
**Pull request by AndersBillLinden**
Not letting scrolling of ScriptsPane change when carrying a block over it. https://github.com/LLK/scratch-flash/issues/742

### CODE REVIEW

**Comment by thisandagain** [comment]
Resolves GH-742
Resolves GH-1080

**Review by cwillisf — APPROVED** [review]

**Comment by TheLogFather** [comment]
Oh, err... Not sure I'm happy about this...

I presume this means we can no longer scroll the pane to see the right side of long blocks? (See https://github.com/LLK/scratch-flash/issues/1080.)

Can we find a temporary alternative to this patch? Maybe have the block palette still scroll when hovering over left/right edges, but have it 'snap back' automatically when the block gets dropped? (Another temporary possibility might be a left/right scrollbar at the bottom?)

**Comment by cwillisf** [comment]
@TheLogFather this change just makes it so that dragging a block near the edge of the block palette won't accidentally scroll the palette area. People often do that accidentally (#742) and "get stuck" because they don't realize how it happened or how to scroll back.

Things like shift+scroll wheel still scroll the palette, but that's only used by a few advanced users like yourself. I agree that we need a better solution like being able to move the divider between the palette and the scripting area or some smarter scrolling system, but that's a larger change than just this one. I suggest using #1080 as a place to talk about that part of the problem.

Note that I've left #1080 open since I don't think this change sufficiently addresses that problem, though I agree with @AndersBillLinden that it fixes #742.

**Comment by TheLogFather** [comment]
Yes, I understood the problem (accidental scroll while hovering at edge during drag – it causes a 'surprise', and such potentially disconcerting behaviour is best avoided), and that the patch only affects drag.

However, shift-scroll appears to work only on certain browsers (works on FF, but not Safari for me).

That's why I suggested the 'snap back' as a potential alternative temporary 'band-aid' fix for the accidental scrolling 'surprise' – yes, the scroll 'surprise' still happens during the drag, but its effects are immediately mitigated on drop.

--

This isn't a particularly big issue – it'll just be a little irritating not having quick access to the right side of longer custom blocks in the "More blocks" palette from within Safari. (Then again, maybe there's something I'm missing in Safari that'll make shift-scroll work...? And any reasonably complicated project I would tend to work on offline with my own compiled version of Scratch that includes all the nice search features. –Though that can be a bit tedious if the project has several cloudvars, since it turns them all into normal globals when saving, and I have to then edit the JSON to switch "persistent" back to true... :/ )

**Comment by liam4** [comment]
> However, shift-scroll appears to work only on certain browsers (works on FF, but not Safari for me).

Doesn't work on any browsers for me (OS X 10.11.6) :(

File: src/util/GestureHandler.as [hunk 1]
```diff
@@ -133,10 +133,12 @@ public class GestureHandler {
 			}
 		}
 		if (carriedObj && scrollTarget && (getTimer() - scrollStartTime) > SCROLL_MSECS && (scrollXVelocity || scrollYVelocity)) {
-			scrollTarget.contents.x = Math.min(0, Math.max(-scrollTarget.maxScrollH(), scrollTarget.contents.x + scrollXVelocity));
-			scrollTarget.contents.y = Math.min(0, Math.max(-scrollTarget.maxScrollV(), scrollTarget.contents.y + scrollYVelocity));
-			scrollTarget.constrainScroll();
-			scrollTarget.updateScrollbars();
+			if (!scrollTarget is ScriptsPane) {
```
Comment (TheLogFather) [inline src/util/GestureHandler.as:136]: Doesn't "ScriptsPane" refer to the whole of the scripts pane?
So scrolling (by dragging block near edge) no longer works anywhere in the scripts pane in v450...?
See https://scratch.mit.edu/discuss/topic/220438/
(Oops!)

Comment (cwillisf) [reply src/util/GestureHandler.as:136]: Yep -- looks like you're right. Thanks for bringing this to my attention!
I'll see if I can find a quick fix. If not I'll revert this change for now.

Comment (cwillisf) [reply src/util/GestureHandler.as:136]: Here's my attempt to fix both #742 without breaking drag-scrolling in the scripting area: #1207

Comment (TheLogFather) [reply src/util/GestureHandler.as:136]: Nice! - much more generic solution. :)

**Pull request merged**